net.sourceforge.vrapper.core/src/net/sourceforge/vrapper/vim/commands/motions/ParenthesesMove.java

```diff
@@ -38,31 +38,39 @@
     //default match algorithm, find match for character under cursor
     @Override
     protected int destination(int offset, TextContent content, int count) throws CommandExecutionException {
-        LineInformation info = content.getLineInformationOfOffset(offset);
+        LineInformation line = content.getLineInformationOfOffset(offset);
         ParenthesesPair pair = null;
-        int index;
-        for(index=offset; index<info.getEndOffset(); index++) {
+
+        // Check for one-character pairs.
+        int startIndex = offset;
+        if (offset == line.getEndOffset() && line.getLength() > 0) {
+            // Special case in visual mode: if the cursor is past the end of
+            // line (on top of the line break), we should start looking at the
+            // character before.
+            startIndex--;
+        }
+        for (int index = startIndex; index < line.getEndOffset(); index++) {
             String c = content.getText(index, 1);
             if (PARENTHESES.containsKey(c)) {
                 pair = PARENTHESES.get(c);
-                break;
+                return findMatch(index, pair, content, count);
             }
         }
-        //it isn't a simple one-character pair, look for alternatives
+
+        // Check for a block comment.
         if (pair == null) {
```

Comment:
You changed the logic in the previous `for` loop so `pair` will now always be null here.  We might as well remove this `if` check.